In [1]:
import pandas as pd
from scipy import stats

df = pd.read_csv("Bank_of_America_Cleaned.csv")

df["Date"] = pd.to_datetime(df["Date"], utc=True)
df = df.sort_values("Date").reset_index(drop=True)

print("Rows and columns:", df.shape)
print("Columns:", df.columns.tolist())
df.head()

Rows and columns: (13329, 11)
Columns: ['Date', 'Open', 'High', 'Low', 'Close', 'Volume', 'ticker', 'name', 'Zero_Volume_Flag', 'Year', 'Close_Outlier_Flag']


,Date,Open,High,Low,Close,Volume,ticker,name,Zero_Volume_Flag,Year,Close_Outlier_Flag
0,1973-02-21 05:00:00+00:00,1.507640,1.507640,1.507640,1.507640,99200,BAC,Bank of America Corporation (BAC) Historical Data,False,1973,False
1,1973-02-22 05:00:00+00:00,1.512735,1.512735,1.512735,1.512735,47200,BAC,Bank of America Corporation (BAC) Historical Data,False,1973,False
2,1973-02-23 05:00:00+00:00,1.507640,1.507640,1.507640,1.507640,133600,BAC,Bank of America Corporation (BAC) Historical Data,False,1973,False
3,1973-02-26 05:00:00+00:00,1.507640,1.507640,1.507640,1.507640,24000,BAC,Bank of America Corporation (BAC) Historical Data,False,1973,False
4,1973-02-27 05:00:00+00:00,1.507640,1.507640,1.507640,1.507640,41600,BAC,Bank of America Corporation (BAC) Historical Data,False,1973,False


In [2]:
df["Daily_Return"] = df["Close"].pct_change(fill_method=None)
returns = df["Daily_Return"].dropna()

print("Observations:", len(returns))
print("Mean daily return:", f"{returns.mean():.6%}")
print("Median daily return:", f"{returns.median():.6%}")
print("Standard deviation:", f"{returns.std():.6%}")

Observations: 13328
Mean daily return: 0.054831%
Median daily return: 0.000000%
Standard deviation: 2.363429%


In [3]:
test_result = stats.ttest_1samp(returns, popmean=0)

print("t-statistic:", test_result.statistic)
print("p-value:", test_result.pvalue)
print("Degrees of freedom:", len(returns) - 1)

t-statistic: 2.678344033028823
p-value: 0.0074077927118944625
Degrees of freedom: 13327


In [4]:
import numpy as np

n = len(returns)
mean_return = returns.mean()
standard_error = returns.std(ddof=1) / np.sqrt(n)

confidence_interval = stats.t.interval(
    confidence=0.95,
    df=n - 1,
    loc=mean_return,
    scale=standard_error
)

print("95% confidence interval:", confidence_interval)


95% confidence interval: (np.float64(0.0001470302221017487), np.float64(0.0009495902352407155))


## Hypothesis Test: Is the Mean Daily Return Different from Zero?

**Null hypothesis (H₀):** The mean daily return is zero.  
**Alternative hypothesis (H₁):** The mean daily return is different from zero.

A two-sided one-sample t-test was performed at the 5% significance level (α = 0.05) using 13,328 daily returns. The mean daily return was 0.054831%. The test produced a t-statistic of 2.6783 and a p-value of 0.0074.

Since the p-value is below 0.05, the null hypothesis is rejected. The 95% confidence interval for the mean daily return was approximately 0.0147% to 0.0950%, which does not include zero. This suggests that the historical mean daily return in this dataset was positive and statistically different from zero.

**Limitation:** This test uses historical closing-price returns. It does not predict future performance, and the test assumes observations are independent. Daily financial returns can be related over time, so the result should be interpreted with caution.